validator/client/runner.go

```diff
@@ -288,24 +296,23 @@ func handleAssignmentError(err error, slot primitives.Slot) {
 	}
 }
 
-func runHealthCheckRoutine(ctx context.Context, v iface.Validator) {
+func runHealthCheckRoutine(ctx context.Context, v iface.Validator, eventsChan chan<- *event.Event) {
 	healthCheckTicker := time.NewTicker(time.Duration(params.BeaconConfig().SecondsPerSlot) * time.Second)
+	tracker := v.NodeHealthTracker(ctx)
 	go func() {
-		for {
-			select {
-			case <-healthCheckTicker.C:
-				if v.NodeIsHealthy(ctx) && !v.EventStreamIsRunning() {
-					if err := v.StartEventStream(ctx); err != nil {
-						log.WithError(err).Error("Could not start API event stream")
-					}
-				}
-			case <-ctx.Done():
-				if ctx.Err() != nil {
-					log.WithError(ctx.Err()).Error("Context cancelled")
-				}
-				log.Error("Context cancelled")
+		// trigger the healthcheck immediately the first time
+		for ; true; <-healthCheckTicker.C {
+			if ctx.Err() != nil {
+				log.WithError(ctx.Err()).Error("Context cancelled")
 				return
 			}
+			newStatus := v.NodeIsHealthy(ctx)
+			tracker.UpdateNodeHealth(newStatus)
+			// in case of node returning healthy but event stream died
+			if !v.EventStreamIsRunning() {
+				log.Info("Event stream reconnecting...")
+				go v.StartEventStream(ctx, event.DefaultEventTopics, eventsChan)
+			}
```

Comment:
```suggestion
			if newStatus && !v.EventStreamIsRunning() {
				log.Info("Event stream reconnecting...")
				go v.StartEventStream(ctx, event.DefaultEventTopics, eventsChan)
			}
```
Might be better to replace the variable name with something like `isHealthy` because `if newStatus` doesn't sound too good

Reply:
cool i added it and changed the name.